public/app/plugins/datasource/loki/LogContextProvider.ts

```diff
@@ -218,17 +207,28 @@ export class LogContextProvider {
     return expr;
   };
 
-  getInitContextFiltersFromLabels = async (labels: Labels) => {
-    await this.datasource.languageProvider.start();
-    const allLabels = this.datasource.languageProvider.getLabelKeys();
+  getInitContextFiltersFromLabels = async (labels: Labels, query?: LokiQuery) => {
+    if (!query || !labels) {
```

Comment:
Can `labels` be falsy? If so, we should fix the type here.

Reply:
It should be `isEmpty()`. Nice catch!